searchcore/src/vespa/searchcore/proton/server/documentdbconfigmanager.cpp

```diff
@@ -344,6 +350,31 @@ DocumentDBConfigManager::update(const ConfigSnapshot &snapshot)
         }
         newRankingConstants = std::make_shared<RankingConstants>(constants);
     }
+    if (snapshot.isChanged<RankingExpressionsConfig>(_configId, currentGeneration)) {
+        RankingExpressionsConfigSP newRankingExpressionsConfig = RankingExpressionsConfigSP(
+            snapshot.getConfig<RankingExpressionsConfig>(_configId));
+        const vespalib::string &spec = _bootstrapConfig->getFiledistributorrpcConfig().connectionspec;
+        RankingExpressions expressions;
+        if (spec != "") {
+            config::RpcFileAcquirer fileAcquirer(spec);
+            vespalib::TimeBox timeBox(5*60, 5);
+            for (const RankingExpressionsConfig::Expression &rc : newRankingExpressionsConfig->expression) {
+                vespalib::string filePath;
+                LOG(info, "Waiting for file acquirer (name='%s', ref='%s')",
```

Comment:
Lines 363 to 372 are almost identical to similar code handling ranking constants and onnx models. Consider adding a helper function to reduce code duplication.

Reply:
fixed